pcss-criminal-application/src/main/java/ca/bc/gov/open/pcsscriminalapplication/service/impl/AppearanceValidatorImpl.java

```diff
@@ -0,0 +1,332 @@
+package ca.bc.gov.open.pcsscriminalapplication.service.impl;
+
+import ca.bc.gov.open.pcsscriminalapplication.service.AppearanceValidator;
+import ca.bc.gov.open.pcsscriminalapplication.utils.ValidationUtils;
+import ca.bc.gov.open.wsdl.pcss.one.*;
+import ca.bc.gov.open.wsdl.pcss.secure.one.GetAppearanceCriminalApprMethodSecureRequest;
+import ca.bc.gov.open.wsdl.pcss.secure.one.GetAppearanceCriminalCountSecureRequest;
+import ca.bc.gov.open.wsdl.pcss.secure.one.GetAppearanceCriminalSecureRequest;
+import org.springframework.stereotype.Service;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+@Service
+public class AppearanceValidatorImpl implements AppearanceValidator {
+
+    @Override
+    public List<String> validateGetAppearanceCriminal(GetAppearanceCriminalRequest getAppearanceCriminalRequest) {
+
+        if (getAppearanceCriminalRequest == null) return Collections.singletonList("Empty request is invalid");
+
+        List<String> errors = new ArrayList<>();
+
+        if (ValidationUtils.checkRequestAgencyIdentifierIdForErrors(getAppearanceCriminalRequest.getRequestAgencyIdentifierId())) {
+            errors.add("RequestAgencyIdentifierId is not valid");
+        }
+
+        if (ValidationUtils.checkSystemIdTypeForErrors(getAppearanceCriminalRequest.getRequestPartId())) {
+            errors.add("RequestPartId is not valid");
+        }
+
+        if (getAppearanceCriminalRequest.getRequestDtm() == null || ValidationUtils.checkDateTimeTypeForErrors(getAppearanceCriminalRequest.getRequestDtm().toString())) {
+            errors.add("RequestDtm is not valid");
+        }
+
+        if (ValidationUtils.checkSystemIdTypeForErrors(getAppearanceCriminalRequest.getAppearanceId())) {
+            errors.add("AppearanceId is not valid");
+        }
+
+        if (ValidationUtils.checkJustinNoTypeForErrors(getAppearanceCriminalRequest.getJustinNo())) {
+            errors.add("JustinNo is not valid");
+        }
```

Comment:
AppearanceId and JustinNo are optional fields, so if one or the other is null, it actually shouldn't be recognized as an error here. There are some other optional fields in the other methods in this file that this applies to as well.

However, as we just investigated, it seems there's some additional validation that checks if **both** of these fields null and throws an error. We will need to add that validation rule at some point as well.